codecs/include/mgs/codecs/basic_codec.hpp

```diff
@@ -138,6 +138,34 @@ class basic_codec
   {
     return output_traits<T>::create(make_decoder(std::forward<Args>(args)...));
   }
+
+  template <typename T = default_encoded_output>
+  static auto lazy_encode()
+  {
+    return
+        [](auto&&... args)
+            -> codecs::codec_output<
+                decltype(output_traits<T>::create(
+                    make_encoder(std::forward<decltype(args)>(args)...))),
+                decltype(make_encoder(std::forward<decltype(args)>(args)...))> {
```

Comment:
Wow, isn't the auto deduction enough? Do you really need all this?

Reply:
Maybe the SFINAE friendliness is not needed, as I can't decltype the lambda untill C++20 (I think).